lib/gringotts/gateways/pin_pay.ex

```diff
@@ -0,0 +1,423 @@
+    commit(:post, "cards", card_for_token(card, opts), opts)
+  end
+
+  @doc """
+  Removes card or payment info that was previously `store/2`d
+
+  Deletes previously stored payment-source data.
+
+  ## Note
+
+  > If there's anything noteworthy about this operation, it comes here.
+
+  ## Example
+
+  > A barebones example using the bindings you've suggested in the `moduledoc`.
+  """
+  @spec unstore(String.t(), keyword) :: {:ok | :error, Response}
+  def unstore(registration_id, opts) do
+    # can't be implemented in pinpayments
+  end
+
+  ###############################################################################
+  #                                PRIVATE METHODS                              #
+  ###############################################################################
+
+  # Makes the request to PinPay's network.
+  # For consistency with other gateway implementations, make your (final)
+  # network request in here, and parse it using another private method called
+  # `respond`.
+
+  defp card_params(card, opts) do
+    [
+      "card[number]": card.number,
+      "card[name]": card.first_name <> card.last_name,
+      "card[expiry_month]": card.month |> Integer.to_string() |> String.pad_leading(2, "0"),
+      "card[expiry_year]": card.year |> Integer.to_string(),
+      "card[cvc]": card.verification_code,
+      "card[address_line1]": opts[:address].street1,
+      "card[address_city]": opts[:address].city,
+      "card[address_country]": opts[:address].country
+    ]
+  end
+
+  defp card_for_token(card, opts) do
+    [
+      "number": card.number,
+      "name": card.first_name <> card.last_name,
+      "expiry_month": card.month |> Integer.to_string() |> String.pad_leading(2, "0"),
+      "expiry_year": card.year |> Integer.to_string(),
+      "cvc": card.verification_code,
+      "address_line1": opts[:Address][:street1],
+      "address_city": opts[:Address][:city],
+      "address_country": opts[:Address][:country]
+    ]
+  end
+
+  @spec commit(atom, String.t(), keyword, keyword) :: {:ok | :error, Response}
+  defp commit(:post, endpoint, param, opts) do
+    auth_token = encoded_credentials("c4nxgznanW4XZUaEQhxS6g", "")
+
+    headers = [
+      {"Content-Type", "application/x-www-form-urlencoded"},
+      {"Authorization", auth_token}
+    ]
+
+    url = @test_url <> "#{endpoint}"
+
+    url
+    |> HTTPoison.post({:form, param}, headers)
+    |> respond
+  end
+
+  defp commit(method, url) do
+    auth_token = encoded_credentials("c4nxgznanW4XZUaEQhxS6g", "")
+
+    headers = [
+      {"Content-Type", "application/x-www-form-urlencoded"},
+      {"Authorization", auth_token}
+    ]
+
+    HTTPoison.request(method, url, [], headers)
+    |> respond
+  end
+
+  defp encoded_credentials(login, password) do
+    [login, password]
+    |> join_string(":")
+    |> Base.encode64()
+    |> (&("Basic " <> &1)).()
+  end
+
+  defp join_string(list_of_words, joiner), do: Enum.join(list_of_words, joiner)
```

Comment:
Use `Enum.join` instead of such helpers -- or name your helpers better, though the latter _is always better_.